### PR TITLE
Remove elements with NoSourcePosition

### PR DESCRIPTION
**Pull request by MaximoOliveira**
See issue #301 

It is better to remove these elements before entering in the loop

Also removed some javadoc that was incorrect in one file

### CODE REVIEW

File: src/main/java/fr/inria/astor/core/solutionsearch/AstorCoreEngine.java [hunk 1]
```diff
@@ -509,7 +508,7 @@ protected void undoSingleGeneration(ProgramVariant instance, int genI) {
 	 * GenOperationInstance
 	 * 
 	 * @param variant
-	 * @param operationofGen
```
Comment (MaximoOliveira) [inline src/main/java/fr/inria/astor/core/solutionsearch/AstorCoreEngine.java:512]: incorrect javadoc

File: src/main/java/fr/inria/astor/core/solutionsearch/population/ProgramVariantFactory.java [hunk 2]
```diff
@@ -59,16 +61,15 @@ public ProgramVariantFactory(List<TargetElementProcessor<?>> processors) {
...
+	 *
 	 * @param suspiciousList
 	 * @param maxNumberInstances
-	 * @param populationControler
 	 * @param projectFacade
 	 * @return
 	 * @throws Exception
 	 */
 	public List<ProgramVariant> createInitialPopulation(List<SuspiciousCode> suspiciousList, int maxNumberInstances,
-			PopulationController populationControler, ProjectRepairFacade projectFacade) throws Exception {
```
Comment (MaximoOliveira) [inline src/main/java/fr/inria/astor/core/solutionsearch/population/ProgramVariantFactory.java:71]: parameter not used in this method

File: src/main/java/fr/inria/astor/core/solutionsearch/AstorCoreEngine.java (hunk 1 above)
Comment (martinezmatias) [reply src/main/java/fr/inria/astor/core/solutionsearch/AstorCoreEngine.java:512]: Great, thanks!

File: src/main/java/fr/inria/astor/core/solutionsearch/population/ProgramVariantFactory.java (hunk 2 above)
Comment (martinezmatias) [reply src/main/java/fr/inria/astor/core/solutionsearch/population/ProgramVariantFactory.java:71]: Perfect thanks :)

**Comment by martinezmatias** [comment]
Hi @MaximoOliveira 

Nice work, thanks a lot. It seems fine for me.

Just one small detail: the PR also contains commits that we have already merged: bee42f6  `* Fix non-deterministic bug that is found with Cardumen.`
As we will squash and merge, that would not be a problem now. 

Thanks! 
Matias

**Pull request merged**